assets/src/application/ui/input/withLabel/index.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+interface Props {
+	id?: string;
+	label?: string;
+}
+
+const withLabel = (Component: JSX.Element) => ({ id, label }: Props): JSX.Element => {
+	return (
+		<div className={'ee-base-input'}>
+			{label && (
+				<label className='ee-base-input-label' htmlFor={id}>
+					{label}
+				</label>
+			)}
+			{Component}
+		</div>
+	);
+};
```

Comment:
That `Component` is a bit misleading there, its rendered as `{Component}` which means it's a node not a component.

```suggestion
const withLabel = (output: React.ReactNode): React.FC<Props> => {
	const Component: React.FC<Props> = ({ id, label } ) => {
		return (
			<div className={'ee-base-input'}>
				{label && (
					<label className='ee-base-input-label' htmlFor={id}>
						{label}
					</label>
				)}
				{output}
			</div>
		);
	}

	return Component;
};
```

Reply:
as I wrote, I am not :100: happy with using HOC in this specific case